File: yyxarch/src/main/java/com/yyxnb/yyxarch/annotation/XGravity.java
package com.yyxnb.yyxarch.annotation;

import android.support.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@IntDef({
        XGravity.CENTER,
        XGravity.LEFT,
        XGravity.RIGHT,
        XGravity.ALIGN_LEFT,
        XGravity.ALIGN_RIGHT,
})
@Retention(RetentionPolicy.SOURCE)
public @interface XGravity {
    int CENTER = 0;
    int LEFT = 1;
    int RIGHT = 2;
    int ALIGN_LEFT = 3;
    int ALIGN_RIGHT = 4;
}

File: yyxarch/src/main/java/com/yyxnb/yyxarch/annotation/XGravity.kt
package com.yyxnb.yyxarch.annotation

import android.support.annotation.IntDef

import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

@IntDef(XGravity.CENTER, XGravity.LEFT, XGravity.RIGHT, XGravity.ALIGN_LEFT, XGravity.ALIGN_RIGHT)
@Retention(RetentionPolicy.SOURCE)
annotation class XGravity {
    companion object {
        const val CENTER = 0
        const val LEFT = 1
        const val RIGHT = 2
        const val ALIGN_LEFT = 3
        const val ALIGN_RIGHT = 4
    }
}